hikari/impl/rest.py

```diff
@@ -696,6 +699,10 @@ async def _request(
         headers.put(_X_AUDIT_LOG_REASON_HEADER, reason)
 
         url = compiled_route.create_url(self._rest_url)
+
+        backoff = None
```

Comment:
To aid people reading the code and linters, might be best to document this and add typehints
```suggestion
        # This is initiated the first time we hit a 5xx error to save memory when nothing goes wrong
        backoff: typing.Optional[rate_limits.ExponentialBackoff] = None
```

Reply:
Alright, committing this